cmd/restic/cmd_check.go

```diff
@@ -352,6 +351,30 @@ func runCheck(ctx context.Context, opts CheckOptions, gopts GlobalOptions, args
 		}
 	}
 
+	filterBySnapshot := false
+	if len(args) > 0 || !opts.SnapshotFilter.Empty() {
+		snapshotLister, err := restic.MemorizeList(ctx, repo, restic.SnapshotFile)
+		if err != nil {
+			return err
+		}
+
+		visitedTrees := restic.NewIDSet()
+		for sn := range FindFilteredSnapshots(ctx, snapshotLister, repo, &opts.SnapshotFilter, args) {
```

Comment:
Should directly use `(&opts.SnapshotFilter).FindAll(...)` instead. The error handling of `FindFilteredSnapshots` is pretty much non-existent.